src/aiokatcp/sensor.py

```diff
@@ -54,13 +54,31 @@
     overload,
 )
 
+try:
+    from typing import Protocol
+except ImportError:
+    # In Python 3.7 it's elsewhere. Mypy after 3.7 doesn't like this.
+    from typing_extensions import Protocol  # type: ignore
+
 from . import core
 
 _T = TypeVar("_T")
+_U = TypeVar("_U")
+
+
+class ChangeAwareObserver(Protocol[_T]):
```

Comment:
I'm not wild about that name. All observers are aware of changes - that's their purpose. How about `OldReadingObserver`? Once we've bikeshedded it obviously all the corresponding variable names need to be changed too.

Reply:
`OldReadingObserver` is explicit but a bit clunky. I thought of `DeltaObserver`?